lib/src/database/matrix_sdk_database.dart

```diff
@@ -1489,7 +1486,7 @@ class MatrixSdkDatabase extends DatabaseApi {
     try {
       await clear();
       await open();
-      final json = Map.from(jsonDecode(export)).cast<String, Map>();
+      final json = Map<String, Map>.from(jsonDecode(export));
```

Comment:
Same here. This does not prevent type errors or makes anything more type safe. It just creates a non-deep copy